### PR TITLE
PostProcessing for Track new plots 2nd PR Trial + ITSTrackCheck vertexZ update

### PR DESCRIPTION
**Pull request by delico18**
New trending plots(mean and stddev) update 

EtaDistribution 
PhiDistribution 
VertexZ 
NVertexContributors
AssociatedClusterFraction
Ntracks
VertexX
VertexY

ITSTrackCheck
- update on vertexZ monitoring condition

### CODE REVIEW

File: Modules/ITS/itsQCTrendingTracks.json [hunk 1]
```diff
@@ -44,6 +71,62 @@
...
+          {
+            "names": ["AssociatedClusterFraction_mean", "AssociatedClusterFraction_stddev"],
+            "title": ["Average AssociatedClusterFraction trend", "Stddev AssociatedClusterFraction trend"],
+            "varexp": ["AssociatedClusterFraction.mean:ntreeentries", "AssociatedClusterFraction.stddev:ntreeentries"],
+            "selection": "",
+            "option": "PL"
+          },
+          {
+            "names": ["Ntracks_mean", "Ntracks_stddev"],
+            "title": ["Average Ntracks trend", "Stddev Ntracks trend"],
```
Comment (iravasen) [inline Modules/ITS/itsQCTrendingTracks.json:112]: In the title put: `Average NTracks per event trend` and `Stddev of NTracks per event`

File: Modules/ITS/itsQCTrendingTracks.json [hunk 2]
```diff
@@ -44,6 +71,62 @@
...
+            "varexp": ["VertexX.sumwx/VertexX.sumw:ntreeentries", "TMath::Sqrt((VertexX.sumwx2/VertexX.sumw)-(VertexX.sumwx/VertexX.sumw)*(VertexX.sumwx/VertexX.sumw)):ntreeentries"],
+            "selection": "",
+            "option": "PL"
+          },
+          {
+            "names": ["VertexY_mean", "VertexY_stddev"],
+            "title": ["Average VertexY trend", "Stddev VertexY trend"],
+            "varexp": ["VertexY.sumwy/VertexY.sumw:ntreeentries", "TMath::Sqrt((VertexY.sumwy2/VertexY.sumw)-(VertexY.sumwy/VertexY.sumw)*(VertexY.sumwy/VertexY.sumw)):ntreeentries"],
+            "selection": "",
+            "option": "PL"
```
Comment (iravasen) [inline Modules/ITS/itsQCTrendingTracks.json:129]: These 2 you can if you define 1 plot in the paths. See the other comment

File: Modules/ITS/src/TrendingTaskITSTracks.cxx [hunk 3]
```diff
@@ -140,17 +140,70 @@ void TrendingTaskITSTracks::storePlots(repository::DatabaseInterface& qcdb)
...
+        class1 = 0;
+        ymin = -15.;
+        ymax = 15.;
+      } else if (plot.name.find("VertexY") != std::string::npos) {
+        class1 = 2;
+        ymin = -15.;
+        ymax = 15.;
+      }
+
     } else if (plot.name.find("stddev") != std::string::npos) {
```
Comment (iravasen) [inline Modules/ITS/src/TrendingTaskITSTracks.cxx:185]: now you have `VertexXY` so you need to merge the two. Please put also `ymin = -1`, `ymax=1`.

File: Modules/ITS/src/TrendingTaskITSTracks.cxx [hunk 4]
```diff
@@ -140,17 +140,70 @@ void TrendingTaskITSTracks::storePlots(repository::DatabaseInterface& qcdb)
...
+      } else if (plot.name.find("AssociatedClusterFraction") != std::string::npos) {
+        class1 = 1;
+      } else if (plot.name.find("Ntracks") != std::string::npos) {
+        class1 = 1;
+      } else if (plot.name.find("VertexX") != std::string::npos) {
+        class1 = 1;
+      } else if (plot.name.find("VertexY") != std::string::npos) {
+        class1 = 3;
+      }
       ymin = 0.;
```
Comment (iravasen) [inline Modules/ITS/src/TrendingTaskITSTracks.cxx:205]: now you have `VertexXY` so you have to merge the two.

File: Modules/ITS/src/TrendingTaskITSTracks.cxx [hunk 5]
```diff
@@ -140,17 +140,70 @@ void TrendingTaskITSTracks::storePlots(repository::DatabaseInterface& qcdb)
...
+    double ymin = 0.;
+    double ymax = 1.;
     if (plot.name.find("mean") != std::string::npos) {
-      add = 0;
-      ymin = 0.;
-      ymax = 15.;
+      if (plot.name.find("NCluster") != std::string::npos) {
+        class1 = 0;
+        ymin = 0.;
+        ymax = 20.;
```
Comment (iravasen) [inline Modules/ITS/src/TrendingTaskITSTracks.cxx:150]: `ymax = 15`

File: Modules/ITS/src/TrendingTaskITSTracks.cxx [hunk 6]
```diff
@@ -140,17 +140,70 @@ void TrendingTaskITSTracks::storePlots(repository::DatabaseInterface& qcdb)
...
+        ymin = 0.;
+        ymax = 2 * TMath::TwoPi();
+      } else if (plot.name.find("VertexZ") != std::string::npos) {
+        class1 = 0;
+        ymin = -15.;
+        ymax = 15.;
+      } else if (plot.name.find("NVertexContributors") != std::string::npos) {
+        class1 = 0;
+        ymin = 0.;
+        ymax = 100.;
```
Comment (iravasen) [inline Modules/ITS/src/TrendingTaskITSTracks.cxx:166]: `ymax = 50`

File: Modules/ITS/src/TrendingTaskITSTracks.cxx [hunk 7]
```diff
@@ -174,12 +174,12 @@ void TrendingTaskITSTracks::storePlots(repository::DatabaseInterface& qcdb)
         ymax = 100.;
       } else if (plot.name.find("VertexX") != std::string::npos) {
```
Comment (iravasen) [inline Modules/ITS/src/TrendingTaskITSTracks.cxx:175]: why don't you put a single `else if (plot.name.find("VertexXY"))`. As said in my previous comment, you don't have anymore VertexX and VertexY but only VertexXY.

File: Modules/ITS/src/TrendingTaskITSTracks.cxx (hunk 3 above)
Comment (delico18) [reply Modules/ITS/src/TrendingTaskITSTracks.cxx:185]: mConfig.plots (not mConfig.datasources) is used here.
so, would merging be appropriate?

File: Modules/ITS/src/TrendingTaskITSTracks.cxx (hunk 7 above)
Comment (delico18) [reply Modules/ITS/src/TrendingTaskITSTracks.cxx:175]: mConfig.plots (not mConfig.datasources) is used here.
so, would merging be appropriate?

Comment (iravasen) [reply Modules/ITS/src/TrendingTaskITSTracks.cxx:175]: you are right. Stupid me.

**Review by iravasen — APPROVED** [review]

**Pull request merged**